react/components/molecules/search/search.js

```diff
@@ -0,0 +1,155 @@
+import React, { PureComponent } from "react";
+import { Animated, StyleSheet, View, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+
+import { ButtonIcon, TextArea } from "../../atoms";
+
+export class Search extends PureComponent {
+    static get propTypes() {
+        return {
+            value: PropTypes.string,
+            placeholder: PropTypes.string,
+            buttonProps: PropTypes.object,
+            onFocus: PropTypes.func,
+            onValue: PropTypes.func,
+            onBlur: PropTypes.func,
+            onClear: PropTypes.func,
+            onSubmit: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            value: undefined,
+            placeholder: "Search",
+            buttonProps: {},
+            onFocus: () => {},
+            onValue: value => {},
+            onBlur: () => {},
+            onClear: () => {},
+            onSubmit: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: this.props.value,
+            buttonsVisible: true,
+            focused: false
+        };
+    }
+
+    setValue = (value, callback = () => {}) => this.setState({ value: value }, callback);
+
+    focus = () => {
+        this.textAreaComponent.focus();
+    };
+
+    blur = () => {
+        this.textAreaComponent.blur();
+    };
+
+    sendMessage = () => {
+        this.props.onSendMessage(this.state.value);
+    };
+
+    clear = () => {
+        const emptyValue = "";
+        this.setState({ value: emptyValue }, () => this.props.onValue(emptyValue));
+        this.focus();
+        this.props.onClear();
+    };
+
+    onTextAreaValue = value => {
+        this.setState({ value: value });
+        this.props.onValue(value);
+    };
+
+    onTextAreaFocus = () => {
+        this.setState({ focused: true }, () => {
+            this.props.onFocus();
+        });
+    };
+
+    onTextAreaBlur = () => {
+        this.setState({ focused: false }, () => {
+            this.props.onBlur();
+        });
+    };
+
+    onTextAreaSubmit = () => {
+        this.setState({ focused: false }, () => {
+            this.props.onSubmit();
+        });
+    };
+
+    onButtonPress = () => {

```

Comment:
Is this normal in applications? I don't remember having to tap de search icon to submit a search, normally the search is made each time we type or remove a character. Its does not make sense to submit an empty text as well. The `search` in  `ripe-components-vue` does not have this behaviour as well.

I think you can remove this (the onSubmit behaviour), and maybe even use `Input` instead of `TextArea` (although I don't know if it behaves exactly the same). What do you think?

Reply:
we need this to clear the text when clicking the `close` icon